spec/controllers/signatures_controller_spec.rb

```diff
@@ -3,87 +3,143 @@
+      it "should redirect to the petition signed page" do
+        get :verify, :id => signature.id, :token => signature.perishable_token
+        expect(assigns[:signature]).to eq(signature)
+        expect(response).to redirect_to("https://petition.parliament.uk/petitions/#{signature.petition_id}/signature/signed")
       end
 
       it "should not set petition state to validated" do
-        get :verify, :id => @signature.id, :token => @signature.perishable_token
-        expect(Petition.find(@signature.petition.id).state).to eq(Petition::PENDING_STATE)
+        get :verify, :id => signature.id, :token => signature.perishable_token
+        expect(petition.reload.state).to eq(Petition::PENDING_STATE)
       end
 
-      it "should set creator signature state to validated and set token to nil" do
-        get :verify, :id => @signature.id, :token => @signature.perishable_token
-        signature = Signature.find(@signature.id)
+      it "should set state to validated and set token to nil" do
+        get :verify, :id => signature.id, :token => signature.perishable_token
+        signature.reload
         expect(signature.state).to eq(Signature::VALIDATED_STATE)
         expect(signature.perishable_token).to be_nil
       end
 
       it "should raise exception if id not found" do
         expect do
-          get :verify, :id => @signature.id + 1, :token => @signature.perishable_token
+          get :verify, :id => signature.id + 1, :token => signature.perishable_token
         end.to raise_error(ActiveRecord::RecordNotFound)
       end
 
       it "should raise exception if token not found" do
         expect do
-          petition = FactoryGirl.create(:petition)
-          @signature = FactoryGirl.create(:signature, :petition => petition, :state => Signature::PENDING_STATE)
-          get :verify, :id => @signature.id, :token => "#{@signature.perishable_token}a"
+          get :verify, :id => signature.id, :token => "#{signature.perishable_token}a"
         end.to raise_error(ActiveRecord::RecordNotFound)
       end
     end
 
-    context "signature to be verified is petition creator's" do
-      before :each do
-        @petition = FactoryGirl.create(:petition)
-        @signature = @petition.creator_signature
+    context 'signature to be verified is a sponsor' do
+      let(:petition) { FactoryGirl.create(:petition) }
+      let(:sponsor) { FactoryGirl.create(:sponsor, petition: petition) }
+      let(:signature) { sponsor.create_signature(FactoryGirl.attributes_for(:pending_signature, petition: petition)) }
+
+      it "should redirect to the petition sponsored page" do
+        get :verify, :id => signature.id, :token => signature.perishable_token
+        expect(assigns[:signature]).to eq(signature)
+        expect(response).to redirect_to("https://petition.parliament.uk/petitions/#{petition.id}/sponsors/#{sponsor.perishable_token}/sponsored")
```

Comment:
Is there a reason you create the assertion URLs by hand instead of using some route helper?